pkg/agent/ipassigner/responder/ndp_responder.go

```diff
@@ -0,0 +1,230 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package responder
+
+import (
+	"fmt"
+	"net"
+	"sync"
+
+	"github.com/mdlayher/ndp"
+	"golang.org/x/net/ipv6"
+	"k8s.io/klog/v2"
+	utilnet "k8s.io/utils/net"
+)
+
+const (
+	solicitedNodeMulticastAddressPrefix = "ff02::1:ff00:0"
+)
+
+type ndpConn interface {
+	WriteTo(message ndp.Message, cm *ipv6.ControlMessage, dstIP net.IP) error
+	ReadFrom() (ndp.Message, *ipv6.ControlMessage, net.IP, error)
+	JoinGroup(net.IP) error
+	LeaveGroup(net.IP) error
+	Close() error
+}
+
+func parseIPv6SolicitedNodeMulticastAddress(ip net.IP) (net.IP, int) {
+	group := net.ParseIP(solicitedNodeMulticastAddressPrefix)
+	// copy lower 24 bits
+	copy(group[13:], ip[13:])
```

Comment:
Is it expected to update ip inplace, which would affect the original IP used in IPAssigner?

Reply:
the dst to copy is the IP parsed above. The original IP will not be changed.

Reply:
I get it. My bad. Thought the opposite.